setup.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+# Basic setup.py so tk-core could be installed as
+# a standard Python package
+from setuptools import setup, find_packages
+import os
+import subprocess
+
+def get_version():
+    """
+    Helper to extract a version number for this module.
+
+    :returns: A major.minor.patch[.sub] version string or "dev".
+    """
+    # Try to extract the version number from git
+    # this will work when installing from a locally cloned repo
+    try:
+        version_git = subprocess.check_output(["git", "describe"]).rstrip()
+        return version_git
+    except:
+        # Blindly ignore problems, git might be not available, or the user could
+        # be installing from zip archive, etc...
+        pass
+
+    # If everything fails, return a sensible string highlighting that the version
+    # couldn't be extracted. If a version is not specified in `setup`, 0.0.0
+    # will be used by default, it seems better to have an explicit keyword for
+    # this case.
+    return "dev"
```

Comment:
this looks brilliant. One comment - i think we should say something other than "dev" in the case it's not coming from git. Toolkit reports `HEAD` in the case the version is not known - I suggest we use that for consistency. It's hopefully a short term thing. Then, when we add proper git releases support to the app store, we'll make sure to get the proper version in even for non-git-describe workflows. How does that sound?